src/mscorlib/corefx/System/Globalization/DateTimeFormatInfo.cs

```diff
@@ -2797,6 +2789,7 @@ void InsertHash(TokenHashValue[] hashTable, String str, TokenType tokenType, int
                                     }
                                 }
                                 // The token to be inserted is already in the table.  Skip it.
+                                return;
```

Comment:
It looks like a bug. The duplicates added will never be found on lookup, except for being skipped when looking for other tokens (which takes time). 
FWIW the hashtable doesn't have buckets but instead a simple hash on the first character of the token, which is then modified if the token is not found in order to loop through all items. That means searching for something that starts with an 'a' may need to skip over items that start with a 'b' for example (when there is more than 1 'a' and more than 1 'b').